plugins/modules/source_control/gitlab/gitlab_project_mirror.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2021, Max-Florian Bidlingmaier (Max-Florian.Bidlingmaier@sap.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: gitlab_project_mirror
+short_description: Creates/updates/deletes GitLab Project Remote Mirror
+description:
+  - When the remote mirror does not exist for the project, it will be created.
+  - When changes are made to the remote mirror, the mirror will be updated.
+  - Gitlab API does not allow to delete a remote mirror as of July 2021. There is an request upstream
+    (https://gitlab.com/gitlab-org/gitlab/-/issues/287786)
+  - Gitlab API does not allow to change (parts) of the URL. As credentials are part of it they can not be checked/changed,
+    this limits idempotency.
+author:
+  - Max-Florian Bidlingmaier (@suukit) Max-Florian.Bidlingmaier@sap.com
+version_added: '3.7.0'
+
+requirements:
+  - python >= 2.7
+  - python-gitlab python module
+extends_documentation_fragment:
+- community.general.auth_basic
+
+options:
+    api_token:
+        description:
+            - GitLab token for logging in.
+        type: str
+    validate_certs:
+        description:
+            - Whether or not to validate TLS/SSL certificates when supplying a HTTPS endpoint.
+            - Should only be set to C(false) if you can guarantee that you are talking to the correct server
+              and no man-in-the-middle attack can happen.
+        default: true
+        type: bool
+    api_username:
+        description:
+            - The username to use for authentication against the API.
+        type: str
+    api_password:
+        description:
+            - The password to use for authentication against the API.
+        type: str
+    project:
+        description:
+            - The name of the project.
+        required: true
+        type: str
+    enabled:
+        description:
+            - Is this remote mirror currently enabled?
```

Comment:
I would formulate these differently (same below):
```suggestion
            - Whether this remote mirror is currently enabled.
```